src/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,175 @@
+    ? 'pagination__icons--primary'
+    : 'pagination__icons--secondary';
+};
+
+// regular pagination
+const BasePagination: FC<IPaginationProps> = (props) => {
+  const {
+    totalPage,
+    className,
+    style,
+    currentPage,
+    onPrev,
+    onNext,
+    size,
+    color,
+    shape,
+    disabled = false,
+    itemTypes,
+    updateCurrentPage,
+  } = usePagination(props);
+
+  let eclipsed = 0;
+
+  let classes = 'pagination';
+  if (className) {
+    classes += ' ';
+  }
+
+  const disabledPrevButton = disabled || currentPage === 1;
+  const disabledNextButton = disabled || currentPage === totalPage;
+
+  return (
+    <div style={style} className={classes} data-testid={'pagination'}>
+      <div>
+        <button
+          data-testid={'prev-btn'}
+          onClick={onPrev}
+          disabled={disabledPrevButton}
+          className={`${getBasicIconClass()} ${getArrowIconClass()} ${getSizeClass(
+            size
+          )} ${getShapeClass(shape)} ${getDisabledClass(disabledPrevButton)}`}
+        >
+          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+            <path d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z" />
+          </svg>
+        </button>
+      </div>
+      <div className={'pagination__icons'}>
+        {itemTypes.map((n, i) => {
+          if (n === 'eclipsed') {
+            eclipsed += 1;
+            return eclipsed > 1 ? null : (
+              <div
+                key={`pagination_number_${i}`}
+                className={`${getDisabledClass(disabled)}`}
+              >
+                {'...'}
+              </div>
+            );
+          } else {
+            eclipsed = 0;
+            if (+i === +currentPage) {
+              return (
+                <div
+                  key={`pagination_number_${i}`}
+                  className={`${getBasicIconClass()} 
+                    ${getSizeClass(size)} ${getShapeClass(
+                    shape
+                  )} ${getColorClass(color)} ${getDisabledClass(
+                    disabled
+                  )} ${getColorFocusedClass()}`}
+                >
+                  {i}
+                </div>
+              );
+            } else {
+              return (
+                <button
+                  key={`pagination_number_${i}`}
+                  disabled={disabled}
+                  onClick={(_) => updateCurrentPage(+i)}
+                  className={`${getBasicIconClass()} ${getSizeClass(
+                    size
+                  )} ${getShapeClass(shape)} ${getColorClass(
+                    color
+                  )} ${getDisabledClass(disabled)}`}
+                >
+                  <div>{i}</div>
+                </button>
+              );
+            }
+          }
+        })}
+      </div>
+      <div>
+        <button
+          data-testid={'next-btn'}
+          onClick={onNext}
+          disabled={disabledNextButton}
+          className={`${getBasicIconClass()} ${getArrowIconClass()}
```

Comment:
Please checkout the util function in the util folder. Or try out third party library to compose the classname